src/replacer.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import { API3 } from "./api"
+
+declare var NodeFilter: any
+
+export class Replacer {
+	__api: API3
+	__idRegex: RegExp = /\b([di]\d{6}|c\d{7})\b/gi
+
+	constructor(api: API3) {
+		this.__api = api
+	}
+}
+
+export class NodeReplacer extends Replacer {
+	_restrictors: Array<(node: Node) => boolean> = []
+
+	restrict(restrictor: (node: Node) => boolean) {
+		this._restrictors.push(restrictor)
+	}
+
+	_checkRestriction(node: Node): boolean {
+		return this._restrictors.reduce((a, b) => a && b(node), true)
+	}
+
+	async watch(element: HTMLElement) {
+		await this._replaceAll(element)
+		const observer = new MutationObserver(this._handleChange.bind(this))
```

Comment:
The previous version of this extension was already using the observer, but the previous version had serious performance issues. So I decided do go with an alternative solution. Maybe combining the observer with a debounce could be a solution. Need to test that.